app/models/artefact.rb

```diff
@@ -156,17 +156,14 @@ def any_editions_ever_published?
   end
 
   def update_editions
-    case state
-    when 'draft'
-      if self.slug_changed?
-        Edition.where(:state.nin => ["archived"],
-                      panopticon_id: self.id).each do |edition| # rubocop:disable Rails/FindEach
-          edition.update_slug_from_artefact(self)
-        end
+    if self.slug_changed?
+      Edition.where(:state.nin => ["archived"],
+                    panopticon_id: self.id).each do |edition| # rubocop:disable Rails/FindEach
+        edition.update_slug_from_artefact(self)
       end
-    when 'archived'
-      archive_editions
     end
+
+    archive_editions if state == 'archived'
```

Comment:
I'm a bit unsure about these changes, as it seems to open the possibility of the slug being changed, at the same time as the edition being archived?

I don't know much about the state machine in Publisher, so this concern might not be reasonable...